spec/lib/annotate/annotate_models_spec.rb

```diff
@@ -2526,44 +2526,256 @@ def annotate_one_file(options = {})
-                               mock_column(:foreign_thing_id, :integer)
-                             ],
+          klass = mock_class(class_name,
+                             primary_key,
+                             original_columns,
                              [],
-                             [
-                               mock_foreign_key('fk_rails_cf2568e89e',
-                                                'foreign_thing_id',
-                                                'foreign_things',
-                                                'id',
-                                                on_delete: :cascade)
-                             ])
-          @schema_info = AnnotateModels.get_schema_info(klass, '== Schema Info', show_foreign_keys: true)
+                             original_foreign_keys)
+          @schema_info = AnnotateModels.get_schema_info(klass, '== Schema Info', options)
           annotate_one_file
         end
 
-        it 'should update foreign key constraint' do
-          klass = mock_class(:users,
-                             :id,
-                             [
-                               mock_column(:id, :integer),
-                               mock_column(:foreign_thing_id, :integer)
-                             ],
-                             [],
-                             [
-                               mock_foreign_key('fk_rails_cf2568e89e',
-                                                'foreign_thing_id',
-                                                'foreign_things',
-                                                'id',
-                                                on_delete: :restrict)
-                             ])
-          @schema_info = AnnotateModels.get_schema_info(klass, '== Schema Info', show_foreign_keys: true)
-          annotate_one_file
+        context 'updating a constraint' do
+          let(:updated_foreign_keys) do
+            [
+              mock_foreign_key('fk_rails_cf2568e89e',
+                               'foreign_thing_id',
+                               'foreign_things',
+                               'id',
+                               on_delete: :restrict)
+            ]
+          end
+
+          context 'when option "format_bare" is true' do
+            let(:format_option) { { format_bare: true } }
+
+            it 'should update foreign key constraint' do
+              klass = mock_class(class_name,
+                                 primary_key,
+                                 original_columns,
+                                 [],
+                                 updated_foreign_keys)
+
+              @schema_info = AnnotateModels.get_schema_info(klass, '== Schema Info', options)
+              annotate_one_file(options)
+
+              expect(File.read(@model_file_name)).to eq("#{@schema_info}#{@file_content}")
+            end
+          end
+
+          context 'when option "format_markdown" is true' do
```

Comment:
There didn't appear to be a test for the markdown format, so this adds a test for `format_markdown` and extracts some of the shared set up into `before`s and `let`s.